gap/base/recognition.gi

```diff
@@ -161,7 +161,20 @@ InstallGlobalFunction( EmptyRecognitionInfoRecord,
         Setisequal(ri,\=);
         ri!.order := Order;
     fi;
-    Setarecommuting(ri,function(x,y) return isequal(ri)(x*y, y*x); end);
+    Setarecommuting(ri, function(x,y)
+      local a,b;
+      if IsObjWithMemory(x) then
+        a := StripMemory(x);
+      else
+        a := x;
+      fi;
+      if IsObjWithMemory(y) then
+        b := StripMemory(y);
+      else
+        b := y;
+      fi;
```

Comment:
@FriedrichRober You can just do this, since `StripMemory` does nothing when its input doesn't have memory.
```suggestion
      a := StripMemory(x);
      b := StripMemory(y);
```

Reply:
You are absolutely right :)